objects/client.c

```diff
@@ -1988,8 +1988,16 @@ client_unmanage(client_t *c, bool window_valid)
     }
 
     if(globalconf.focus.client == c)
+    {
         client_unfocus(c);
 
+        /* Avoid unmanage and unfocus to be in the same lua iteration. This
+         causes a race condition where delayed redraw are registered for an
+         already unmanaged client
+         */
+        awesome_refresh();
+    }
```

Comment:
Doesn't this just make it a lot harder to hit the problem, but it can still happen if awesome just happens to be slow, or something like that? I'm not a fan (but I guess you already knew).